pkgs/by-name/me/meson/package.nix

```diff
@@ -63,6 +64,14 @@ python3.pkgs.buildPythonApplication rec {
     # https://github.com/NixOS/nixpkgs/issues/86131#issuecomment-711051774
     ./005-boost-Do-not-add-system-paths-on-nix.patch
 
+    # https://github.com/NixOS/nixpkgs/issues/86131
+    # upstream PR: https://github.com/mesonbuild/meson/pull/13272
+    (fetchpatch {
+      name = "find-boost-pkg-config.patch";
+      url = "https://github.com/mesonbuild/meson/commit/f0eda0346b2b936e4ef795f3a82733d6711b18fb.patch";
```

Comment:
URLs for unmerged commits are not stable — if you rebase the patch this commit will be garbage collected and not exist, so for longevity this sort of patch needs to be vendored.